resources/linux/debian/postinst.template

```diff
@@ -35,19 +35,42 @@ if [ "@@NAME@@" != "code-oss" ]; then
 	eval $(apt-config shell APT_TRUSTED_PARTS Dir::Etc::trustedparts/d)
 	CODE_TRUSTED_PART=${APT_TRUSTED_PARTS}microsoft.gpg
 
+	RET=true
+	if [ -e '/usr/share/debconf/confmodule' ]; then
+		. /usr/share/debconf/confmodule
+		db_get @@NAME@@/add-microsoft-repo || true
+	fi
+
 	# Install repository source list
 	WRITE_SOURCE=0
-	if [ ! -f $CODE_SOURCE_PART ] && [ ! -f /etc/rpi-issue ]; then
-		# Write source list if it does not exist and we're not running on Raspberry Pi OS
-		WRITE_SOURCE=1
-	elif grep -Eq "http:\/\/packages\.microsoft\.com\/repos\/vscode" $CODE_SOURCE_PART; then
+	if [ "$RET" false ]; then
+		# The user does not want to add the microsoft repository
+		WRITE_SOURCE=0
+	elif grep -q "http://packages.microsoft.com/repos/vscode" $CODE_SOURCE_PART; then
 		# Migrate from old repository
 		WRITE_SOURCE=1
+	elif apt-cache policy | grep -q "https://packages.microsoft.com/"; then
```

Comment:
I made this change do to your message
> Would `grep -q https://packages.microsoft.com` work instead?
> When I run `apt-cache policy | grep https://packages.microsoft.com` I get two results that both contain `https://packages.microsoft.com/ubuntu/22.04/prod` rather than /repos/code.

Should I change it back? Personally, I would say the original was more correct since the current code simply adds https://packages.microsoft.com/repos/code either way. Alternatively, we could check if there is any repo providing code, although that might lead to someone being stuck with a distro repo that provides an older version...

Reply:
What happened was I had an HTTP link to `/repos/code`, so after running your grep command, I thought that the script added a link to Microsoft via `https://packages.microsoft.com/ubuntu/22.04/prod`, which is not correct.
I recommend changing it back.